libs/SalesforceSDK/src/com/salesforce/androidsdk/config/AbstractPrefsManager.java

```diff
@@ -140,13 +142,28 @@ public void resetAll() {
         final String sharedPrefPath = SalesforceSDKManager.getInstance().getAppContext().getApplicationInfo().dataDir + "/shared_prefs";
         final File dir = new File(sharedPrefPath);
         final FilenameFilter fileFilter = new FilenameFilter() {
+
             @Override
             public boolean accept(File dir, String filename) {
                 return filename != null && filename.startsWith(getFilenameRoot());
             }
         };
         for (final File file : dir.listFiles()) {
             if (file != null && fileFilter.accept(dir, file.getName())) {
+
+                // Removes extension of filename before feeding it to SharedPreferences.
+                String filename = file.getName();
+                filename = filename.substring(0, filename.lastIndexOf('.'));
+
+                /*
+                 * Removes the SharedPreferences object explicitly before removing the
+                 * underlying file. The OS caches the contents of SharedPreferences files
+                 * in memory for fast retrieval. If we don't explicitly remove it here,
+                 * the next write to the same file will result in stale data being re-added.
+                 */
+                final SharedPreferences sp = SalesforceSDKManager.getInstance().getAppContext().getSharedPreferences(filename,
+                        Context.MODE_PRIVATE);
+                sp.edit().clear().commit();
```

Comment:
<table data-meta="generated_by_SalesforceSDK">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/3863d488116759db0c0ffeed7e2a323048531e82/libs/SalesforceSDK/src/com/salesforce/androidsdk/config/AbstractPrefsManager.java#L166"></span>Consider using <code>apply()</code> instead; <code>commit</code> writes its data to persistent storage immediately, whereas <code>apply</code> will handle it in the background</td>
    </tr>
  </tbody>
</table>